controller/engine_controller.go

```diff
@@ -2119,25 +2122,30 @@ func (ec *EngineController) Upgrade(e *longhorn.Engine, log *logrus.Entry) (err
 		err = errors.Wrapf(err, "failed to live upgrade image for %v", e.Name)
 	}()
 
-	engineClientProxy, err := ec.getEngineClientProxy(e, e.Spec.Image)
-	if err != nil {
-		return err
-	}
-	defer engineClientProxy.Close()
-
-	version, err := engineClientProxy.VersionGet(e, false)
-	if err != nil {
-		return err
-	}
+	if types.IsDataEngineV1(e.Spec.DataEngine) {
+		engineClientProxy, err := ec.getEngineClientProxy(e, e.Spec.Image)
+		if err != nil {
+			return err
+		}
+		defer engineClientProxy.Close()
 
-	// Don't use image with different image name but same commit here. It
-	// will cause live replica to be removed. Volume controller should filter those.
-	if version.ClientVersion.GitCommit != version.ServerVersion.GitCommit {
-		log.Infof("Upgrading engine from %v to %v", e.Status.CurrentImage, e.Spec.Image)
-		if err := ec.UpgradeEngineInstance(e, log); err != nil {
+		version, err := engineClientProxy.VersionGet(e, false)
+		if err != nil {
 			return err
 		}
+
+		// Don't use image with different image name but same commit here. It
+		// will cause live replica to be removed. Volume controller should filter those.
+		if version.ClientVersion.GitCommit != version.ServerVersion.GitCommit {
+			log.Infof("Upgrading engine from %v to %v", e.Status.CurrentImage, e.Spec.Image)
+			if err := ec.UpgradeEngineInstance(e, log); err != nil {
+				return err
+			}
+		}
+	} else {
+		return errors.Wrapf(err, "failed to upgrade engine %v with data engine %v", e.Name, e.Spec.DataEngine)
```

Comment:
It seems that this operation is not supported by v2 engine, not that it is failed.

Reply:
Updated. Thank you.